### PR TITLE
Enable ability to edit and view in Review and Submit section.

### PR DESCRIPTION
**Pull request by awong-dev**
connected to
https://github.com/department-of-veterans-affairs/healthcare-application/issues/103

### CODE REVIEW

**Comment by awong-dev** [comment]
Just a hack, but at least it gets us somewhere.

**Comment by awong-dev** [comment]
FYI, I won't be able to get back to this for a few days so feel free to just take the code and discard, run with it, or do whatever necessary to get the functionality in.

**Comment by crwallace** [comment]
This includes:
- Edit/Update link that will toggle the users ability to edit any information that is not correct in the review section.
- Added a `sectionComplete` prop that will allow components in <ReviewAndSubmitSection/> and <Nav/> to know when the section has been completed.
- Styles for review and edit modes.

This is what the user will see when they arrive to <ReviewAndSubmitSection/>:
screen shot 2016-04-04 at 2 53 35 pm

This is what the user will see when they click "Edit" in <ReviewAndSubmitSection/>:
screen shot 2016-04-04 at 2 53 57 pm

File: _health-care/_js/components/financial-assessment/AnnualIncomeSection.jsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import { connect } from 'react-redux';
 
 import ErrorableTextInput from '../form-elements/ErrorableTextInput';
+import ErrorableCheckbox from '../form-elements/ErrorableCheckbox';
```
Comment (awong-dev) [inline _health-care/_js/components/financial-assessment/AnnualIncomeSection.jsx]: Alphabetical order within this block please :)

File: _health-care/_js/components/financial-assessment/ChildInformationSection.jsx [hunk 2]
```diff
@@ -64,6 +64,11 @@ class ChildInformationSection extends React.Component {
     return (
       <div>
         <h4>Children Information</h4>
+        <ErrorableCheckbox
+            label={`${this.props.data.sectionComplete ? 'Edit' : 'Update'}`}
+            checked={this.props.data.sectionComplete}
+            className={`edit-checkbox ${this.props.reviewSection ? '' : 'hidden'}`}
+            onValueChange={(update) => {this.props.onStateChange('sectionComplete', update);}}/>
```
Comment (awong-dev) [inline _health-care/_js/components/financial-assessment/ChildInformationSection.jsx]: The idea here is on review, there is no edit/update button?

**Comment by awong-dev** [comment]
LGTM w/ nits to add comments on new section properties.

**Comment by akainic** [comment]
@crwallace @awong-dev Sorry I'm late to the party, but I'm a little concerned about this implementation. Styling inputs so they don't look or behave like inputs doesn't seem very semantic, nor great for accessibility. I understand why we may not have wanted to add additional markup in the review section for the label and value of each question (e.g., something to handle the display of `First name: Test`), and then conditionally rendering the section component if the user clicks the `Edit` link, but it seems much more semantic and clear to have separate markup for what is really 2 different functionalities (displaying a summary of submitted information vs. updating that information). We could put conditionals in the sections themselves that determine what markup is rendered for a "review" state and what is rendered for an "edit" state, since some markup will be shared between the two. This could make it a bit DRYer, but also potentially far more confusing. Any thoughts on these options vs. other alternatives?

**Comment by awong-dev** [comment]
I defer to you two on this.

Organizing markup is not something I have experience with.
Naively, I would have wanted some sort of generic <Question/> component
that would modally be either in "review" or "edit" mode and internally
switch between an `<input>` field and another sort of DOM element. That way
the *Section components (which, btw...probably should be renamed *Panel)
only reads like a list of questions.

However, I'm not sure if that conflicts with the semantic organization you
are proposing...

**Comment by akainic** [comment]
Yes, I think that could work as well. That's along the lines of what I was thinking, of rendering either an input or some other DOM element based on what mode the component is in. Hadn't thought of creating a new `<Question/>` component though, that's interesting. One concern is the number of props we would have to pass to that component in order for each individual question to have the information it needs, since each question takes such different information based on the type of input and the format of the data.

**Comment by awong-dev** [comment]
https://facebook.github.io/react/docs/transferring-props.html  <-- Object
Rest Spread syntax help?

**Pull request merged**